engines/macventure/macventure.cpp

```diff
@@ -0,0 +1,1184 @@
+			break;
+		case 0x8:
+			reflectSwap(obj.object, obj.target);
+			break;
+		case 0xc:
+			_world->setObjAttr(_gui->getWindowData(kMainGameWindow).refcon, kAttrContainerOpen, 0);
+			_world->setObjAttr(_world->getObjAttr(1, kAttrParentObject), kAttrContainerOpen, 1);
+			break;
+		case 0xd:
+			toggleExits();
+			break;
+		case 0xe:
+			zoomObject(obj.object);
+			break;
+		}
+	}
+}
+
+void MacVentureEngine::printTexts() {
+	for (uint i = 0; i < _textQueue.size(); i++) {
+		QueuedText text = _textQueue.front();
+		_textQueue.remove_at(0);
+		switch (text.id) {
+		case kTextNumber:
+			_gui->printText(Common::String(text.asset));
+			gameChanged();
+			break;
+		case kTextNewLine:
+			_gui->printText(Common::String(""));
+			gameChanged();
+			break;
+		case kTextPlain:
+			_gui->printText(_world->getText(text.asset, text.source, text.destination));
+			gameChanged();
+			break;
+		}
+	}
+}
+
+void MacVentureEngine::playSounds(bool pause) {
+	int delay=0;
+	while (!_soundQueue.empty()) {
+		QueuedSound item = _soundQueue.front();
+		_soundQueue.remove_at(0);
+		switch (item.id) {
+		case kSoundPlay:
+			_soundManager->playSound(item.reference);
+			break;
+		case kSoundPlayAndWait:
+			delay = _soundManager->playSound(item.reference);
+			break;
+		case kSoundWait:
+			//wait for sound to finish?
+			break;
+		}
+	}
+	if (pause && delay > 0) {
+		warning("Sound pausing not yet tested. Pausing for %d", delay);
+		g_system->delayMillis(delay);
+		preparedToRun();
+	}
+}
+
+void MacVentureEngine::updateControls() {
+	selectControl(kNoCommand);
+	_gui->clearControls();
+	toggleExits();
+	resetVars();
+}
+
+void MacVentureEngine::resetVars() {
+	selectControl(kNoCommand);
+	_currentSelection.clear();
+	_destObject = 0;
+	setDeltaPoint(Common::Point(0, 0));
+	_cmdReady = false;
+}
+
+void MacVentureEngine::unselectAll() {
+	while (!_currentSelection.empty()) {
+		unselectObject(_currentSelection.front());
+	}
+}
+
+void MacVentureEngine::selectObject(ObjID objID) {
+	if (!_currentSelection.empty()) {
+		if (findParentWindow(objID) != findParentWindow(_currentSelection[0])) {
+			//unselectAll();
+		}
+	}
+	if (findObjectInArray(objID, _currentSelection) == -1) {
+		_currentSelection.push_back(objID);
+		highlightExit(objID);
+	}
+}
+
+void MacVentureEngine::unselectObject(ObjID objID) {
+	int idxCur = findObjectInArray(objID, _currentSelection);
+	if (idxCur != -1){
+		_currentSelection.remove_at(idxCur);
+		highlightExit(objID);
+	}
+}
+
+
+void MacVentureEngine::updateExits() {
+	_gui->clearExits();
+	_gui->unselectExits();
+
+	Common::Array<ObjID> exits = _world->getChildren(_world->getObjAttr(1, kAttrParentObject), true);
+	for (uint i = 0; i < exits.size(); i++)
+		_gui->updateExit(exits[i]);
+
+}
+
+int MacVentureEngine::findObjectInArray(ObjID objID, const Common::Array<ObjID> &list) {
+	// Find the object in the current selection
+	bool found = false;
+	uint i = 0;
+	while (i < list.size() && !found) {
+		if (list[i] == objID) found = true;
```

Comment:
You are using way too many one line statements. Please consider rearranging the code.